File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.cpp
#include <string>
#include <vector>
#include <sstream>
#include <map>
#include <set>

using namespace std;

typedef struct node {
    vector<string> user;    //나를 신고한 사용자
    int count;      //결과값
    node() {
        vector<string> v;
        this->user = v;
        count = 0;
    }
}NODE;

vector<int> solution(vector<string> id_list, vector<string> report, int k) {
    vector<int> answer;
    map<string, NODE> count;
    for (int i = 0; i < id_list.size(); i++)
        count.insert({id_list[i], node()});

    set<string> S;
    for (int i = 0; i < report.size(); i++) {
        if (S.find(report[i]) != S.end())
            continue;
        S.insert(report[i]);
        stringstream str(report[i]);
        string ing, ed;
        str >> ing >> ed;
        count[ed].user.emplace_back(ing);
    }
    
    for (int i = 0; i < id_list.size(); i++) {
        if (count[id_list[i]].user.size() < k)
            continue;
        for (int j = 0; j < count[id_list[i]].user.size(); j++) {
            string name = count[id_list[i]].user[j];
            count[name].count++;
        }
    }
    
    for (int i = 0; i < id_list.size(); i++)
        answer.emplace_back(count[id_list[i]].count);
    return answer;
}

File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.swift
import Foundation

struct info {
    var idx: Int
    var users: [String]
    init(_ idx: Int) {
        self.idx = idx
        self.users = [String]()
    }
}

func solution(_ id_list:[String], _ report:[String], _ k:Int) -> [Int] {
    var answer = Array(repeating: 0, count: id_list.count)
    var dict: [String: info] = [:]  //value: 나를 신고한 유저 목록
    id_list.enumerated().forEach { dict[$0.element] = info($0.offset) }
    
    report.forEach {
        let rep = $0.components(separatedBy: " ")
        if !dict[rep[1]]!.users.contains(rep[0]) { dict[rep[1]]?.users.append(rep[0]) }
    }
    
    id_list.forEach {
        if dict[$0]!.users.count >= k {
            dict[$0]!.users.forEach { answer[dict[$0]!.idx] += 1 }
        }
    }
    return answer
}